src/extraction/FStar.Extraction.ML.Term.fs

```diff
@@ -1205,12 +1205,12 @@ and term_as_mlexpr' (g:uenv) (top:term) : (mlexpr * e_tag * mlty) =
     //precondition: is_match head = true
     let apply_to_match_branches head args =
       match (head |> SS.compress |> U.unascribe).n with
-      | Tm_match (scrutinee, _, branches) ->
+      | Tm_match (scrutinee, _, branches, _) ->
         let branches =
           branches |> List.map (fun (pat, when_opt, body) ->
             pat, when_opt, { body with n = Tm_app (body, args) }
           ) in
-        { head with n = Tm_match (scrutinee, None, branches) }  //AR: dropping the return annotation, should be fine?
+        { head with n = Tm_match (scrutinee, None, branches, None) }  //AR: dropping the return annotation, should be fine?
```

Comment:
yes, I guess at this stage the residual comp is no longer relevant, since we're way past needing to retypecheck terms in the fastpath etc